### PR TITLE
Contribution documentation + PR template.

### PR DESCRIPTION
**Pull request by Fuzznimp**

### CODE REVIEW

**Comment by Fenntasy** [comment]
Task linked: [CU-62ke0u Contributing Documentation](https://app.clickup.com/t/62ke0u)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+<!--- here to help! -->
+
+- :red_circle: My code follows the code style of this project.
+- :red_circle: My change requires a change to the documentation.
+- :red_circle: I have updated the documentation accordingly.
+- :red_circle: I have read the [**CONTRIBUTING**][contributing_file] document.
+- :red_circle: I have added tests to cover my changes.
+- :red_circle: All new and existing tests passed.
+
+[contributing_file]: https://github.com/fewlinesco/connect/blob/master/CONTRIBUTING.md
```
Comment (Fenntasy) [inline .github/PULL_REQUEST_TEMPLATE.md:57]: Let's make it the contributing of this repo since `connect-account` is public

File: README.adoc [hunk 2]
```diff
@@ -1,7 +1,23 @@
 = Connect Account
 
+Connect Account is an open source app, which aims at providing all the business logic to handle connect identities and profile data, and a default frontend for user to use it.
```
Comment (Fenntasy) [inline README.adoc:3]:
```suggestion
Connect Account is an open source app, which aims at providing all the business logic to handle Connect identities and profile data, and a default frontend for Users to use it.
```

File: README.adoc [hunk 3]
```diff
@@ -1,7 +1,23 @@
...
+----
+git fetch --upstream
+git rebase --upstream
+git push origin master
+git checkout -b name-of-your-branch
+----
+
+== Forking
+
+
```
Comment (Fenntasy) [inline README.adoc:19]:
```suggestion

```

File: README.adoc [hunk 4]
```diff
@@ -13,30 +29,273 @@ These can be installed via the provided `.tool-versions` using `asdf install`.
...
 source .env_vars
 ----
 
 == Development Process
 
-[source,shell]
+[source, shell]
 ----
-yarn dev
+NODE_TLS_REJECT_UNAUTHORIZED=0 yarn dev
```
Comment (Fenntasy) [inline README.adoc:48]:
```suggestion
yarn dev

# or, if you're working with a local version of Connect
NODE_TLS_REJECT_UNAUTHORIZED=0 yarn dev
```

File: README.adoc [hunk 5]
```diff
@@ -13,30 +29,273 @@ These can be installed via the provided `.tool-versions` using `asdf install`.
...
+- */workflows*: GitHub Actions used to run tests during CI/CD process flow.
+- *PULL_REQUEST_TEMPLATE*: Template used when opening a pull request on GitHub.
+
+=== /src
+
+- *@types/*: Type declaration used in multiple places.
+- *assets/*: Folder containing assets.
+- *commands/*: Write (e.g. `POST`) database actions.
+- *components/*:
+  - *business/*: React components that provide business logic through render props. You shouldn't touch them.
```
Comment (Fenntasy) [inline README.adoc:132]:
```suggestion
  - *business/*: React components that provide business logic through render props. You shouldn't have to touch them.
```

File: README.adoc [hunk 6]
```diff
@@ -13,30 +29,273 @@ These can be installed via the provided `.tool-versions` using `asdf install`.
...
+  - *visualization/*: React component that
```
Comment (Fenntasy) [inline README.adoc:133]: that?

File: README.adoc [hunk 7]
```diff
@@ -13,30 +29,273 @@ These can be installed via the provided `.tool-versions` using `asdf install`.
...
+To understand the flow of `connect-account`, you should read the `connect` https://developers.fewlines.co/guides/authorization_code_flow/overview/[documentation].
+
+[source, jsx]
 ----
+<BusinessComponent>
+  {({ businessLogic }) => (
+    <VisualizationComponent businessLogic={businessLogic} />
+  )}
+</BusinessComponent>
+----
```
Comment (Fenntasy) [inline README.adoc:112]: Shouldn't this be in the `=== Splitting logic from visualization components` section?

File: README.adoc [hunk 8]
```diff
@@ -13,30 +29,273 @@ These can be installed via the provided `.tool-versions` using `asdf install`.
...
+=== /tests
+- *__mocks__*: Centralized mocked data or function.
+- **.test.(ts|tsx)*: Unit test files.
+
+=== config files
+- *package.json*: We use this file, as much as possible as a centralized config file for various packages, like `ESLint`, `Jest` or `Husky`.
+- *tsconfig.json*: TypeScript compiler options.
+- *next.config.js*: Extended webpack compiler config used by NextJS.
+
+=== Splitting logic from visualization components
```
Comment (Fenntasy) [inline README.adoc:152]:
```suggestion
=== Spliting logic from visualization components
```

File: README.adoc [hunk 9]
```diff
@@ -13,30 +29,273 @@ These can be installed via the provided `.tool-versions` using `asdf install`.
...
+
+To help minimized conflict from upstream in your fork, we chose to split the business logic — which is handle by us — from visualization components. To do so, we are using https://reactjs.org/docs/render-props.html#gatsby-focus-wrapper[Render Props]:
```
Comment (Fenntasy) [inline README.adoc:154]:
```suggestion
To help minimized conflict from upstream in your fork, we chose to split the business logic — which is handled by us — from visualization components. To do so, we are using https://reactjs.org/docs/render-props.html#gatsby-focus-wrapper[Render Props]:
```

Comment (Fenntasy) [reply README.adoc:154]:
```suggestion
To help minimized conflict from upstream in your fork, we chose to split the business logic — which is handle by us — from visualization components. To do so, we are using https://reactjs.org/docs/render-props.html[Render Props]:
```

or am I missing something?

File: README.adoc [hunk 10]
```diff
@@ -13,30 +29,273 @@ These can be installed via the provided `.tool-versions` using `asdf install`.
...
+const StyledComponent = styled.div`
+  css-property: ${({ theme }) => theme.themeKey};
+`
+----
+
+You can access the theme properties and values from everywhere else with the custom `useTheme` hook:
+
+[source, typescript]
+----
+import { useTheme } from "relative/path/to/the/hook";
```
Comment (Fenntasy) [inline README.adoc:251]:
```suggestion
import { useTheme } from "relative/path/to/the/hooks/useTheme";
```

where is it by the way?

Comment (Fuzznimp) [reply README.adoc:251]: `src/design-system/theme/useTheme.ts`

Comment (Fuzznimp) [reply README.adoc:251]: I'm still wondering if it should go into `/hooks` though.

Comment (Fenntasy) [reply README.adoc:251]:
```suggestion
import { useTheme } from "relative/path/to/the/design-system/theme/useTheme";
```
then 😛

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (Fuzznimp) [reply .github/PULL_REQUEST_TEMPLATE.md:57]: You're right. I'm also wondering it there is any use for the `CONTIRBUTING.md` file.

Comment (Fenntasy) [reply .github/PULL_REQUEST_TEMPLATE.md:57]: not at the moment lest we split your README in two ^^

File: README.adoc (hunk 6 above)
Comment (Fuzznimp) [reply README.adoc:133]: > React component that only — for the most part — handle the visualization part. 

Not a huge fan but I can't really find better wording.

Comment (Fenntasy) [reply README.adoc:133]: 👍

File: README.adoc (hunk 7 above)
Comment (Fuzznimp) [reply README.adoc:112]: Nice catch, dunno why it went there in the end ahah.

File: README.adoc (hunk 8 above)
Comment (Fuzznimp) [reply README.adoc:152]: [:D](https://www.linguee.fr/anglais-francais/traduction/splitting.html)

File: README.adoc (hunk 9 above)
Comment (Fuzznimp) [reply README.adoc:154]: Woooops

**Comment by Fuzznimp** [comment]
@Fenntasy I made the amends, but I'll also add the new routing when it will be merged.

File: doc/flows.puml [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+account -> connect : Fetch OpenID configuration
+note left
+The **@fwl/oauth2** package mainly provides
+an abstraction for the OAuth2 flow, and a
+secure way to verify JWT for **HS256** and
+**RS256** algorithms
+end note
+connect -> account : Send back OpenID configuration and stores it in-memory
+
+== Connect authentification flow ==
```
Comment (alli5723) [inline doc/flows.puml:52]: I think `authentification` is only valid in French 😁 

```suggestion
== Connect authentication flow ==
```

Comment (Fuzznimp) [reply doc/flows.puml:52]: Aaahhh nice catch haha 😅.

Comment (Fuzznimp) [reply doc/flows.puml:52]: Fixed (and I also fixed the typo in the diagram)
ab81795

File: doc/flows.puml [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+== OAuth2 client initialization ==
+
+account -> connect : Fetch OpenID configuration
+note left
+The **@fwl/oauth2** package mainly provides
+an abstraction for the OAuth2 flow, and a
+secure way to verify JWT for **HS256** and
+**RS256** algorithms
+end note
+connect -> account : Send back OpenID configuration and stores it in-memory
```
Comment (alli5723) [inline doc/flows.puml:50]: Everything looks good to me, but I have one question about this part.
I don't know much about this step. Is this the way this application is able to identify what scopes it requests for? Or why do we need this?

Comment (Fuzznimp) [reply doc/flows.puml:50]: Yes exactly.
I should have added a link to the package I wrote a few weeks ago documentation: https://github.com/fewlinesco/node-web-libraries/tree/master/packages/oauth2

This package is a small part of what will become the `Account JS SDK`.

I'll add a section about this. Thanks for your feedback!

**Review by alli5723 — DISMISSED** [review]

File: README.adoc [hunk 13]
```diff
@@ -1,7 +1,39 @@
...
+Connect Account is an open source app, which aims at providing all the business logic to handle Connect identities and profile data, and a default frontend for User to use it.
 
-## Setup
+== Flows
+
+=== Connect authentication flow
+
+image::./doc/oauth2-flow-diagram.svg[OAuth2 flow diagram]
+
+For more information, check the Authorization code flow https://developers.fewlines.co/guides/authorization_code_flow/overview/[documention].
```
Comment (arnaudmorisset) [inline README.adoc:12]:
```suggestion
For more information, check the Authorization code flow https://developers.fewlines.co/guides/authorization_code_flow/overview/[documentation].
```

Comment (arnaudmorisset) [reply README.adoc:12]: Please, don't tell me that this typo is the real URL. 😅

Comment (Fuzznimp) [reply README.adoc:12]: Haha no, it's just the way `asciidoc` handle hyperlinks.

Comment (Fuzznimp) [reply README.adoc]: a62d1f2 Fixed. If you could re-approved, thanks :).

**Review by arnaudmorisset — APPROVED** [review]

**Pull request merged**